### ISSUE TITLE
AddFontFromFileTTF + GetGlyphRangesChineseFull  does not work?

### ISSUE THREAD

**Issue by Charltsing** [mellinoe/ImGui.NET#283]
ImGui.NET.SampleProgram, i add some code:

ImGuiIOPtr ioptr = ImGui.GetIO();
ioptr.Fonts.AddFontFromFileTTF("c:\\Windows\\Fonts\\ArialUni.ttf", 18.0f, null, ioptr.Fonts.GetGlyphRangesChineseFull());

 string text = "中文abc";         //chinese text
 ImGui.Text(text);

it does not work, why?

***************************************
i use dear imgui-master :

io.Fonts->AddFontFromFileTTF("c:\\Windows\\Fonts\\ArialUni.ttf", 18.0f, NULL, io.Fonts->GetGlyphRangesChineseFull());
ImGui::Text(u8"中文abc");

it is work well.

**Comment by zaafar** [comment]
It works fine on my side. For example implementation try looking at this line: https://github.com/zaafar/ClickableTransparentOverlay/blob/master/Examples/SingleThreadedOverlayWithCoroutines/SampleOverlay.cs

**Comment by Charltsing** [comment]
> It works fine in my project. For example implementation try looking at this line: https://github.com/zaafar/ClickableTransparentOverlay/blob/master/Examples/SingleThreadedOverlayWithCoroutines/SampleOverlay.cs#L88

it does not work in ImGui.NET.SampleProgram

i dont know where insert code ---> io.Fonts.AddFontFromFileTTF...

**Comment by zaafar** [comment]
Yeah SampleProgram isn’t updated to support this feature but that’s fine because it’s just a Sample Program. However, this is a supported feature of ImGui.NET and can be used by the users of ImGui library as demonstrated in the URL I shared.

**Comment by Charltsing** [comment]
> Yeah SampleProgram isn’t updated to support this feature but that’s fine because it’s just a Sample Program. However, this is a supported feature of ImGui.NET library and can be used by the users of ImGui library as demonstrated in the URL I shared.

大哥，你那代码我实在看不下去。而且，调用Fonts.clear之后，Sample的代码会花屏。不调用它，你的代码就不工作。
Could you please add Chinese support to the official Sample?

**Comment by zaafar** [comment]
> Could you please add Chinese support to the official Sample?

what’s the benefit? I have already shared with you an example that works.

**Comment by zaafar** [comment]
Make sure Font.Clear happens after ‘ _gd.SwapBuffers(_gd.MainSwapchain);’ function call. Remember, you can not change fonts while they are being used.

also, look at my https://github.com/zaafar/ClickableTransparentOverlay/blob/master/ClickableTransparentOverlay/ImGuiController.cs file. I changed it. The one in sample program is broken.

**Comment by LonerDo** [comment]
In ImGui.NET.SampleProgram,you should remove the code:
//ImGui.GetIO().Fonts.AddFontDefault();

also,you could choose the font in style editor:
![Snipaste_2021-12-28_16-08-30](https://user-images.githubusercontent.com/38101933/147544508-c25db1b4-3f7b-4fdd-b7b8-e868219bfc6e.png)

it works well.
